context/AnalyticsContext/index.tsx

```diff
@@ -9,14 +9,33 @@ import { networkConfig } from '~/constants/networks';
 import { useStore } from '~/store/main';
 import { selectAccount, selectNetwork } from '~/store/Web3Slice';
 
-const writeKey = process.env.NEXT_PUBLIC_SEGMENT_WRITE_KEY_DEV || '';
 const POOLS_VERSION = 2;
 
 // Helper functions
 const convertBNToFloat = (bn: BigNumber) => {
     return parseFloat(BigNumber.max(bn).toString());
 };
 
+// const getWriteKey = () => {
+//     switch (process.env.NODE_ENV) {
+//         case 'production':
+//             return process.env.NEXT_PUBLIC_SEGMENT_WRITE_KEY_PROD;
+//         case 'development':
+//             return process.env.NEXT_PUBLIC_SEGMENT_WRITE_KEY_DEV;
+//         default:
+//             return process.env.NEXT_PUBLIC_SEGMENT_WRITE_KEY_DEV;
+//     }
+// };
+
+const getWriteKey = () => {
+    const url = window.location.href;
+    if (url.includes(process.env.siteUrl as string)) {
+        return process.env.NEXT_PUBLIC_SEGMENT_WRITE_KEY_PROD;
+    } else {
+        return process.env.NEXT_PUBLIC_SEGMENT_WRITE_KEY_DEV;
+    }
+};
```

Comment:
were there complications in environment variables on netlify or something? pretty sure we should just have one environment variable `NEXT_PUBLIC_SEGMENT_WRITE_KEY` which we just change the value of between deploy previews/local and on the actual prod site